source/adapters/level_zero/context.hpp

```diff
@@ -333,10 +334,38 @@ struct ur_context_handle_t_ : _ur_object {
         return WithProfiling ? &EventCaches[2] : &EventCaches[3];
       }
     }
+  };
+  auto getCounterBasedEventCache(bool UsingImmediateCmdList,
+                                 ur_device_handle_t Device) {
+    if (UsingImmediateCmdList) {
+      if (Device) {
+        auto EventCachesMap = &EventCachesDeviceMap[4];
+        if (EventCachesMap->find(Device) == EventCachesMap->end()) {
+          EventCaches.emplace_back();
+          EventCachesMap->insert(
+              std::make_pair(Device, EventCaches.size() - 1));
+        }
+        return &EventCaches[(*EventCachesMap)[Device]];
+      } else {
+        return &EventCaches[4];
```

Comment:
why 4? 
can you add common place to control it ?

Reply:
slot 4 is reserved for counterbased immediate